### PR TITLE
Log stdout and stderr from plugin installation and removal

### LINKED ISSUE #1665: Plugins: removing plugin on Windows brings up cmd shell
Hopefully we're just missing a hide windows shell flag.

### LINKED ISSUE #1673: Plugins: capture installation and removal logging
Working on a binary build version with plugins something failed during installation. An error is reported in a modal window of the command that failed, but no other stdout or stderror is captured. It'd be nice to capture this in a separate log file or add it to `main.log`. 

Slightly related to #1653

### PR DESCRIPTION
**Pull request by emlys**
## Description
Fixes #1673 
Fixes #1665 

Stdout and stderr from subprocesses involved in plugin install/uninstall are now logged. They continue to show up in the terminal in dev mode, and now also appear in the workbench log in dev and production mode. Plugin installation errors are no longer displayed in the UI, but a message is shown pointing to the workbench log.

## Checklist
- [ ] Updated HISTORY.rst and link to any relevant issue (if these changes are user-facing)
- [ ] Updated the user's guide (if needed)
- [ ] Tested the Workbench UI (if relevant)

### CODE REVIEW

File: workbench/src/main/setupAddRemovePlugin.js [hunk 1]
```diff
@@ -11,59 +11,98 @@ import { settingsStore } from './settingsStore';
 
 const logger = getLogger(__filename.split('/').slice(-1)[0]);
 
+function spawnWithLogging(cmd, args, options) {
```
Comment (dcdenu4) [inline workbench/src/main/setupAddRemovePlugin.js:14]: Should this function have a docstring above it?

**Review by dcdenu4 — COMMENTED** [review]
Thanks @emlys ! The async chaining approach makes sense to me and reads okay too. I had one documentation question.

I'm also curious, now that this is asynchronous, is the user able to close out of the plugin modal why the process is running? I feel like a common thing might be for a user to want to continue working on other things while the plugin loads and get back to the main Workbench window. Nothing needed to address in this PR, just thinking ahead.

File: workbench/src/main/setupAddRemovePlugin.js (hunk 1 above)
Comment (emlys) [reply workbench/src/main/setupAddRemovePlugin.js:28]: Added a docstring

**Comment by emlys** [comment]
@dcdenu4 it is possible to close the plugin modal and the plugin install keeps running in the background. There are probably some possible UX improvements to that situation but it does seem to work

File: workbench/src/main/setupAddRemovePlugin.js [hunk 2]
```diff
@@ -11,59 +11,112 @@ import { settingsStore } from './settingsStore';
...
         const tmpPluginDir = fs.mkdtempSync(upath.join(tmpdir(), 'natcap-invest-'));
-        execSync(
-          `git clone --depth 1 --no-checkout ${pluginURL} "${tmpPluginDir}"`,
-          { stdio: 'inherit', windowsHide: true }
-        );
-        execSync('git checkout HEAD pyproject.toml', { cwd: tmpPluginDir, stdio: 'inherit', windowsHide: true });
+        await spawnWithLogging(
+          'git',
+          ['clone', '--depth', '1', '--no-checkout', pluginURL, tmpPluginDir]
+        ).then(async () => {
```
Comment (davemfish) [inline workbench/src/main/setupAddRemovePlugin.js:68]: @emlys , forgive me for popping in here. I don't think we need the `.then` chaining now that we're using `await` and resolving the promise when the spawn process closes. I think you should just be able to do like this,

```javascript
await spawnWithLoggin('git clone ...');
await spawnWithLogging('git checkout ...');
...
```

**Review by dcdenu4 — APPROVED** [review]
Thanks @emlys, I'll add my approval since I'll be offline for most of today. But note @davemfish comment!

**Review by davemfish — APPROVED** [review]
Looks good!

**Pull request merged**